src/engagement.ts

```diff
@@ -0,0 +1,183 @@
+  core.info('Running in engagement scoring mode')
+
+  const octokit = github.getOctokit(config.token)
+  const engagementResponse = await calculateEngagementScores(config, octokit)
+  core.info(`Calculated engagement scores for ${engagementResponse.totalItems} items`)
+
+  // Update project with scores if requested
+  if (config.applyScores) {
+    await updateProjectWithScores(config, engagementResponse, octokit)
+  }
+
+  // Save engagement response to file
+  const engagementFile = `${config.tempDir}/engagement-response.json`
+  const fs = await import('fs')
+  await fs.promises.writeFile(engagementFile, JSON.stringify(engagementResponse, null, 2))
+
+  return engagementFile
+}
+
+/**
+ * Calculate engagement scores for issues in a project or single issue
+ * @param config - Configuration object containing project and authentication details
+ * @param octokit - GitHub API client
+ * @returns Promise<EngagementResponse> - The engagement response with scores
+ */
+export async function calculateEngagementScores(
+  config: EngagementConfig,
+  octokit: ReturnType<typeof github.getOctokit>
+): Promise<EngagementResponse> {
+  if (config.projectNumber && config.projectNumber > 0) {
+    return await calculateProjectEngagementScores(config, octokit)
+  } else if (config.issueNumber && config.issueNumber > 0) {
+    return await calculateIssueEngagementScores(config, octokit)
+  } else {
+    throw new Error('Either project number or issue number must be specified')
+  }
+}
+
+/**
+ * Calculate engagement scores for a single issue
+ */
+async function calculateIssueEngagementScores(
+  config: EngagementConfig,
+  octokit: ReturnType<typeof github.getOctokit>
+): Promise<EngagementResponse> {
+  core.info(`Calculating engagement score for issue #${config.issueNumber}`)
+
+  const issueDetails = await getIssueDetails(octokit, config.repoOwner, config.repoName, config.issueNumber!)
+  const score = calculateScore(issueDetails)
+  const previousScore = await calculatePreviousScore(issueDetails)
+
+  const item: EngagementItem = {
+    issueNumber: issueDetails.number,
+    engagement: {
+      score,
+      previousScore,
+      classification: score > previousScore ? 'Hot' : null
+    }
+  }
+
+  return {
+    items: [item],
+    totalItems: 1
+  }
+}
+
+/**
+ * Calculate engagement scores for all issues in a project
+ */
+async function calculateProjectEngagementScores(
+  config: EngagementConfig,
+  octokit: ReturnType<typeof github.getOctokit>
+): Promise<EngagementResponse> {
+  core.info(`Calculating engagement scores for project #${config.projectNumber}`)
+
+  const projectNumber = config.projectNumber!
+  const projectItems = await getAllProjectItems(octokit, config.repoOwner, config.repoName, projectNumber)
+
+  const items: EngagementItem[] = []
+
+  for (const projectItem of projectItems) {
+    if (projectItem.content?.type === 'Issue') {
+      const issueDetails = await getIssueDetails(
```

Comment:
This code looks to be sort of duplicating calculateIssueEngagementScores. Refactor both functions so that we use the same implementation.